### PR TITLE
Smooth l1 type fix

### PR DESCRIPTION
**Pull request by abhudev**
Fix testConsistency issue (smooth l1 was failing for float16 input)

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/LossOps.mm [hunk 1]
```diff
@@ -771,12 +771,12 @@ void smooth_l1_loss_impl(
           MPSGraphTensor *mpsGraphOneTensor = [mpsGraph constantWithScalar: 1.0
                                                                   dataType: inputTensor.dataType];
           MPSGraphTensor *mpsGraphHalfTensor = [mpsGraph constantWithScalar: 0.5
-                                                                   dataType: MPSDataTypeFloat32];
+                                                                   dataType: inputTensor.dataType];
```
Comment (DenisVieriu97) [inline aten/src/ATen/native/mps/operations/LossOps.mm:774]: Will integer data types work with this change?  (e.g if input was torch.int32)?

Comment (abhudev) [reply aten/src/ATen/native/mps/operations/LossOps.mm:774]: Good point, but I think the loss ops only take float inputs - even in test consistency, the values associated with the key "nn.functional.smooth_l1_loss" are only torch.float16 and torch.float32.

**Review by DenisVieriu97 — APPROVED** [review]

**Review by kulinseth — APPROVED** [review]
